src/main/kotlin/org/ionproject/integration/utils/Try.kt

```diff
@@ -32,26 +37,29 @@ sealed class Try<out T> {
         override fun <R> match(success: (T) -> R, error: (Exception) -> R) = success(value)
     }
 
-    class Error(val e: Exception) : Try<Nothing>() {
+    class Error<out T>(val e: Exception) : Try<T>() {
 
-        override fun <R> map(f: (Nothing) -> R): Try<R> = this
-        override fun <R> flatMap(f: (Nothing) -> Try<R>): Try<R> = this
-        override fun <R> match(success: (Nothing) -> R, error: (Exception) -> R) = error(e)
+        override fun get() = throw e
+        override fun <R> map(f: (T) -> R): Try<R> = this as Error<R>
+        override fun mapError(f: (Exception) -> Exception): Try<T> = Error(merge(e, f(e)))
+        override fun <R> flatMap(f: (T) -> Try<R>): Try<R> = this as Error<R>
+        override fun <R> match(success: (T) -> R, error: (Exception) -> R) = error(e)
     }
 
     companion object {
-        fun <T> of(value: T) = Value(value)
-        fun ofError(e: Exception) = Error(e)
         fun <T> of(f: () -> T): Try<T> = try {
             Value(f())
         } catch (e: Exception) {
             Error(e)
         }
 
+        fun <T> ofValue(value: T) = Value(value)
+        fun <T> ofError(e: Exception) = Error<Exception>(e)
```

Comment:
Could this be ... = Error<T>(e), to make sure the Try is specifically tiped to match the type of e?